File: src/my_project/easy_problems/from101to150/delete_columns_make_sorted.py
from typing import List 

class Solution:
    def minDeletionSize(self, strs: List[str]) -> int:

        temp: str = ""
        temp_list: List[str] = list()
        temp_list_ordered: List[str] = list()
        temp_ordered = None
        solution: int = 0
      
        for i in range(len(strs[0])):
            for word in strs:
                temp += word[i]
            temp_list.append(temp)
            temp = ""

        for word in temp_list:
            temp_ordered = [x for x in word]
            temp_ordered.sort()
            temp_ordered = "".join(temp_ordered)
            temp_list_ordered.append(temp_ordered)

        for i in range(len(temp_list)):
            if temp_list[i] != temp_list_ordered[i]:
                solution += 1

            
        return solution


solution = Solution()
print(solution.minDeletionSize(strs = ["zyx","wvu","tsr"]))

File: my_project_typescript/easy_problems_typescript/from1to50/delete_columns_make_sorted.ts
function minDeletionSize(strs: string[]): number {

    let temp: string = "";
    let tempList: Array<string> = []
    let tempListOrdered: Array<string> = []
    let tempOrdered: any = null 
    let solution: number = 0
    let listOrdered: Array<string>= []

    for (let i = 0; i < strs[0].length; i++){
        for (let word of strs){
            temp += word[i]
        }
        tempList.push(temp)
        temp = ""  
    }
    
    for (let word of tempList){
        temp = ""
        tempListOrdered = []
        for (let letter of word){
            tempListOrdered.push(letter)
        }
        tempListOrdered.sort((x,y) => x <= y ? -1 : 1)
        for (let letter of tempListOrdered){
            temp += letter
        }
        listOrdered.push(temp)
    }

    for (let i = 0; i < listOrdered.length; i++){
        if (listOrdered[i] !== tempList[i]){
            solution +=1
        }
    }  
    return solution
};

console.log(minDeletionSize(["a","d","g"]))


